test/verify/check-system-info

```diff
@@ -274,74 +277,218 @@ class TestSystemInfo(MachineCase):
         # Add two NTP servers.  We can't expect the servers to be used, so
         # we only test that they get added.
-        self.set_change_time_dialog_mode("Automatically using specific NTP servers")
+        self.set_change_time_dialog_mode("Automatically using NTP")
+        b.set_checked("#cb-specific", True)
         b.set_input_text("#systime-ntp-servers tr:nth-child(1) input", "0.pool.ntp.org")
-        b.click('#systime-ntp-servers tr:nth-child(1) button:first-child')
+        b.click('#systime-ntp-servers tr:nth-child(1) button.add-server')
         b.set_input_text("#systime-ntp-servers tr:nth-child(2) input", "1.pool.ntp.org")
         b.click("#system_information_change_systime .apply")
         b.wait_not_present("#system_information_change_systime")
 
-        self.assertIn("0.pool.ntp.org", m.execute(f"grep '^NTP=' {conf}"))
-        self.assertIn("1.pool.ntp.org", m.execute(f"grep '^NTP=' {conf}"))
+        assert_in_enabled("0.pool.ntp.org")
+        assert_in_enabled("1.pool.ntp.org")
 
         # restarts timesyncd to pick up the new config
-        wait(lambda: get_timesyncd_start() > prev_timesyncd_start, delay=0.2)
-        prev_timesyncd_start = get_timesyncd_start()
+        wait(lambda: get_daemon_start() > prev_daemon_start, delay=0.2)
+        prev_daemon_start = get_daemon_start()
 
         # Set conf from the outside, check that we pick that up, and
         # switch to default servers.
-        m.write(conf, "[Time]\nNTP=2.pool.ntp.org\n")
+        set_conf("2.pool.ntp.org")
         b.wait_attr("#system_information_systime_button", "data-timedated-initialized", "true")
         b.click("#system_information_systime_button")
         b.wait_visible("#system_information_change_systime")
         b.wait_val("#systime-ntp-servers tr:nth-child(1) input", "2.pool.ntp.org")
-        self.set_change_time_dialog_mode("Automatically using NTP")
+        b.set_checked("#cb-specific", False)
         b.wait_not_present("#systime-ntp-servers")
         b.click("#system_information_change_systime .apply")
-        b.wait_not_present("#system_information_change_systime")
+        with b.wait_timeout(120):  # Changing time on Arch can be slow
+            b.wait_not_present("#system_information_change_systime")
 
-        self.assertIn("2.pool.ntp.org", m.execute(f"grep '^#NTP=' {conf}"))
+        assert_in_disabled("2.pool.ntp.org")
 
         # restarts timesyncd to pick up the new config
-        wait(lambda: get_timesyncd_start() > prev_timesyncd_start, delay=0.2)
+        wait(lambda: get_daemon_start() > prev_daemon_start, delay=0.2)
+
+    @skipImage("chronyd not available", "arch")
+    def testTimeServersChronyd(self):
+        m = self.machine
+        b = self.browser
+
+        if m.image.startswith("debian") or m.image.startswith("ubuntu"):
+            # install chronyd
+            self.addPackageSet("chronyd")
+            self.enableRepo()
+            m.execute("apt-get update && apt-get install -y chrony")
```

Comment:
s/&&/;/ please. `&&` will silently succeed if the `update` operation fails. m.execute() is `set -e`, so `;` DTRT. Same for the command below.

Reply:
I think `false && true` will fail, but I'll change this to use `;`.

Reply:
Yes, but *only* if it's the last command. Any extra command after that will be run. It will work with this specific command line, but using this pattern is still a giant trap, so let's better all forget about it :grin: